src/Common/src/System/Globalization/IdnMapping.Windows.cs

```diff
@@ -20,14 +20,22 @@ private unsafe string GetAsciiCore(char* unicode, int count)
             }
 
             // Do the conversion
-            char[] output = new char[length];
-            length = Interop.mincore.IdnToAscii(flags, unicode, count, output, length);
-            if (length == 0)
+            const int StackAllocThreshold = 512; // arbitrary limit to switch from stack to heap allocation
+            if (length < StackAllocThreshold)
             {
-                ThrowForZeroLength("unicode", SR.Argument_IdnIllegalName, SR.Argument_InvalidCharSequenceNoIndex);
+                char* output = stackalloc char[length];
+                Interop.mincore.IdnToAscii(flags, unicode, count, output, length);
```

Comment:
@hughbe although it is not expected to have the length changed in the second call of (IdnToAscii or IdnToUnicode), but it is important to check the new returned length at least to validate the call is succeeded. the call can fail for other reasons. 

so I suggest you get the returned length and validate it is greater than zero and then we can add assert that the returned length is equal to the previous returned length. also always use the last returned length to allocate the string we return.